shop-the-look/api/v1/endpoints/text.py

```diff
@@ -0,0 +1,68 @@
+import os
+import requests
+from fastapi import APIRouter, HTTPException
+from pydantic import BaseModel
+from api.config import settings
+from api import deps
+
+router = APIRouter()
+
+class TextQuery(BaseModel):
+    query: str
+
+@router.post("/search/text")
+async def query_text(query: TextQuery):
+    try:
+        if not query.query:
+            raise HTTPException(status_code=400, detail="The query text cannot be empty")
+
+        # Get the access token
+        access_token = settings.get_access_token()
+
+        # Prepare the API request
+        url = f"https://{settings.location}-aiplatform.googleapis.com/v1/projects/{settings.project_id}/locations/{settings.location}/publishers/google/models/multimodalembedding@001:predict"
```

Comment:
Same NIT as above re: a utility function for preventing this line getting too long to comfortably read.